sound/soc/sof/sof-client-probes.c

```diff
@@ -23,6 +23,10 @@
 
 #define SOF_PROBES_INVALID_NODE_ID UINT_MAX
 
+static bool __read_mostly sof_probes_enabled;
+module_param_named(enable, sof_probes_enabled, bool, 0444);
+MODULE_PARM_DESC(enable, "Enable SOF probes support");
```

Comment:
is this in addition to or a replacement to SND_SOC_SOF_DEBUG_PROBES?

Or could it be 

static bool __read_mostly sof_probes_enabled = IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_PROBES);

??

Reply:
It is in addition.
Currently if you enable the probes support, it will do nothing as you need to patch the platform code to create the needed compressed link.

I don't think it is a good idea to have that probes compressed around for normal users as it can be confusing and when we will have the real compressed support, it is going to be even more problematic for users.

I want to keep the two levels of gating of this feature, we might in the future opt to compile the probes support w/o user Kconfig option, but that can be done at a later time.

Reply:
If you have both SND_SOC_SOF_DEBUG_PROBES and sof_probes_enabled, then what is the logic to actually enable the probes. is there any precedence between the two?
Usually when we have a Kconfig and a kernel parameter, the Kconfig is used as a default that can be overridden by a kernel parameter. We just had this discussion for the trace.

Reply:
@plbossart, the logic has not been changed much but simplified, imho.
Currently if the `SND_SOC_SOF_DEBUG_PROBES` is enabled the probes will _not_ function. One needs to modify the kernel to add a new DAI link for the platform, recompile the kernel.

We replaced the need for code change and compilation with a flag. I believe that it would be not good to have only one level for the probes. It might be a nice thing if distros ship with it enabled and we can use it if needed, but normal users will be not confused by it.

One thing I'm not too happy about is that I return with 0 from probe if it is not explicitly enabled by the flag to avoid error print about failed module load.

Reply:
@ujfalusi can you please clarify your view of the logic when both the Kconfig option and kernel parameter are available?

with the sof_probes_enabled = IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_PROBES); I suggested above, the module parameter overrides Kconfig selections, what are you suggesting?

Reply:
If CONFIG_SND_SOC_SOF_DEBUG_PROBES is not enabled the probes related code is not compiled.

If the CONFIG_SND_SOC_SOF_DEBUG_PROBES is enabled the code is compiled and the device for the probes client is going to be created on platforms where it is supported (HDA currently).

If a developer really wants to use the probes then additional module parameter is needed:
```
options snd_sof_probes enable=1
```

Reply:
@ujfalusi your logic is problematic because there are two types of 'developers'
a) the hard-core people who change code
b) acoustic engineers or CI folks who want to verify functionality on released images
Requiring a new build to be able to extract PCM samples is a problem, we shouldn't require this. A single option is much better IMHO, it's ok to create a device which does nothing.

Put differently, there is no value in compiling out the probes, is there?

Reply:
@plbossart, with this logic an acoustic engineer would not need to change code and recompile the kernel, what we have atm would require them to understand ASoC macros and how to add new DAI link for the platform they are using.
I would expect that distributions will have the probes enabled and iMX for example can drop the probes code as it is not working on it.

Sure, we can drop the user facing PROBES selection and have the Intel Kconfig select it in case of HDA. Less Kconfig for the user is the better imho.

Reply:
"we can drop the user facing PROBES selection and have the Intel Kconfig select it in case of HDA"

That would still enable the probes even if the target is baytrail for single-builds, but the run-time platform-specific selection would not create the client device.

Reply:
baytrail is not not a HDA, baytrail only kernel should not have PROBES compiled.

I only have user facing option for the probes in case HDA is enabled, I can remove that and have it selected there without condition.

Reply:
The only kernel that's so custom is probably the Yocto/Merrifield stuff. All other distros and our very own kernel config enable all platforms.
Let's go with with direct selection, thanks!

Reply:
@plbossart, OK, let's drop the user facing options for the probes.

Should we consider removing the similar user facing Kconfig for the dma-trace in a similar manner with the enable parameter?
One thing which bugs me with that is the current sof_debug flag to force enable the dma-trace, that is going to be a NOP. But it is a NOP already with the client conversion.